### DISCUSSION TITLE
Optionaly disable OpenMP ...

### DISCUSSION THREAD

**Discussion by kif** [mesonbuild/meson#13226] [Q&A]
First of all, I would like to thank you for `meson`, it is a very nice build system, making the management of dependencies almost transparent in a cross-platform manner. I especially love the:
`omp = dependency('openmp', required : false)`

I am currently switching CI-platform for building the wheels (of a python project, built with `meson-python`) and found the windows packages broken because they were linked with `libgomp.dll` which is not generally available. This means meson finds the OpenMP dependency and compiles the code with it ... and as a consequence, the package is not working elsewhere.
https://github.com/silx-kit/pyFAI/issues/2162
I wonder if you could suggest a way to disable the `omp` dependency even though it could be found ? I thought on an environment variable but maybe there are better ways to do it. An environment variable would be handy since the code has to be executed in a CI environment where `meson` is called from `pip wheel` called from `ci-buildwheels` ... many layers one on top of the other.

Thanks for your advice.  

Jerome Kieffer

**Comment by kif** [comment]
I found the `option` and the `feature` in meson... this allows to setup easily something like this:
`meson -Duse_openmp=disabled`

The problem is that `meson` is embedded in `meson-python` which is called by `pip wheel` called from `cibuildwheel` ... not simple to have the information gone through. 
Since some of PyPA developer impose all configuration to go via environment variable, which is incompatible with the philosophy of `meson` (according to https://github.com/mesonbuild/meson/issues/9#issuecomment-381410972) I had a problem ... 

But finally, going via a configuration file in root of the project looks like a viable solution (to be tested)
https://github.com/silx-kit/pyFAI/pull/2166/files

**Comment by eli-schwartz** [comment]
IIRC you can have a cibuildwheel config file that defines arguments always passed to pip, such as in this case `-C setup-args=-Dopenmp=disabled`. It would then run any time cibuildwheel is used, which of course is what you want in order to produce consistent PyPI wheels.
